### DISCUSSION TITLE
intent of LOSS_SCALE

### DISCUSSION THREAD

**Discussion by hturki** [NVlabs/instant-ngp#453] [Q&A] [answered]
First of all, thanks for the great work. Is there any specific rationale behind setting LOSS_SCALE to 128 in https://github.com/NVlabs/instant-ngp/blob/16212cd9e0c80d8345896a864f5af7054723a1d0/include/neural-graphics-primitives/testbed.h#L228? Or is it just a magic number that worked well in practice?

**Comment by Tom94** [comment, marked as answer]
Hi there, yes, it is a magic number that helps make better use of fp16 numbers during backpropagation through the neural network. There's otherwise no effect on training (the loss is multiplied by `LOSS_SCALE` before backpropagation, and the gradients are afterwards divided by `LOSS_SCALE` to compensate). 

See [this documentation](https://docs.nvidia.com/deeplearning/performance/mixed-precision-training/index.html) for details.

Cheers!
